app/react/Attachments/components/specs/AttachmentsModal.spec.js

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import thunk from 'redux-thunk';
+import { shallow } from 'enzyme';
+import { LocalForm } from 'react-redux-form';
+import ReactModal from 'react-modal';
+import configureMockStore from 'redux-mock-store';
+import { Provider } from 'react-redux';
+
+import { AttachmentsModalCmp } from '../AttachmentsModal';
+
+const mockStore = configureMockStore([thunk]);
+const store = mockStore({});
+
+describe('Attachments Modal', () => {
+  let component;
+  let props;
+
+  beforeEach(() => {
+    props = {
+      entitySharedId: '123',
+      storeKey: '456',
+      onClose: jasmine.createSpy('onClose'),
+      uploadAttachmentFromUrl: jasmine.createSpy('uploadAttachmentFromUrl'),
+    };
+  });
+
+  const render = (otherProps = {}) => {
+    component = shallow(
+      <Provider store={store}>
+        <AttachmentsModalCmp {...props} {...otherProps} />
+      </Provider>
+    ).dive();
+  };
+
+  it('Should pass isOpen props to attachments modal.', () => {
+    render({ isOpen: false });
+    expect(component.find(ReactModal).props().isOpen).toBe(false);
+    render({ isOpen: true });
+    expect(component.find(ReactModal).props().isOpen).toBe(true);
+  });
+
+  it('Should match render of upload form', () => {
+    render();
+
+    expect(component).toMatchSnapshot();
+  });
+
+  it('Should match render of web form', () => {
+    render();
+
+    component.find('.modal-tab-2').simulate('click');
+
+    expect(component).toMatchSnapshot();
```

Comment:
Is it possible to rewrite the test in order to avoid snapshots?

Reply:
yes, but this test is just testing that the correct components are loaded when switching tabs, I think this is the scenario on when snapshots are good, doing a find of the component and spec the props is the reason we introduced snapshots, the problem is it was followed by what we call "The snapshot fever of the mid century"